sdk/servicebus/service-bus/src/util/utils.ts

```diff
@@ -424,3 +424,37 @@ function buildRawAuthorizationRule(authorizationRule: AuthorizationRule): any {
   };
   return rawAuthorizationRule;
 }
+
+export enum EntityStatus {
+  Active = "Active",
+  ReceiveDisabled = "ReceiveDisabled",
+  SendDisabled = "SendDisabled",
+  Disabled = "Disabled",
+  Unknown = "Unknown"
+}
+
+/**
+ *  @ignore
+ * Helper utility to construct `EntityStatus` from given input,
+ * or undefined if not passed in.
+ * @param status
+ */
+export function getEntityStatusOrUndefined(status: any): EntityStatus | undefined {
```

Comment:
If I understand correctly, we are trying to validate user input for status and ensuring that it is one of the 5 known values defined in the enum `EntityStatus`.

What is the service response if a request is made with an invalid status value?
If it throws an error or even ignores the invalid value, then we should be good and shouldnt need to carry out these checks

Reply:
Can possibly remove it.
Since .NET SDK allows only these as the permissible inputs, should we also do the same as such (even if service returns useful error)?
The status can actually be several other values, think there are 8-9 in total. But only these are allowed as input and is also why the type in response/Details is different.

Reply:
A guideline we are trying to follow when validating user input is to avoid doing so if the service is able to handle the input appropriately (either ignore and use a default or throw error)

Reply:
In that case .NET SDK too should have been accepting a string?
On testing, looks like it ignores and takes a default of "Active" even if something invalid is passed - current changes keep parity with .NET and it might have been done so for a reason.
cc: @AlexGhiondea

Reply:
I wonder if the current approach also has an issue when the service starts sending back a new status for this. Since we don't have that value in the enum, this would throw an exception even if the value coming from the service is valid.

Can you share the .NET code that handles this? I wonder how they are handling this (if at all).

Reply:
In JS this isn't an issue as the only validation we do is for the inputs which seem like they can only be specific values.
https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/servicebus/Microsoft.Azure.ServiceBus/src/Management/QueueDescriptionExtensions.cs#L116 is where the result is being parsed, I don't see any specific handling being added

Reply:
Looks like other possible values maybe transient ones used by service only - https://docs.microsoft.com/en-us/dotnet/api/microsoft.servicebus.messaging.entitystatus?view=azure-dotnet

Reply:
Even for input values, what if we want to pass a value that the service knows about and the client library does not? 

In general (in Track 2) we have been trying to avoid doing client side validation for things that the service can change. That being said, in this case, I am not opposed to do the additional validation.

Reply:
If we want to restrict the input values, we can update the type in the user input to be enum instead of a string. Please sync up offline with @bterlson on the right approach here.

Reply:
Sure, the input is an enum currently.

Reply:
We decided to use union of string literals to help TS end users, and remove the runtime validation so as to allow service to handle the inputs as it deems befitting.

Currently, we are seeing the default of "Active" being used for random inputs. 
One concern was if we wanted to specifically disallow any values for any reason or it was just .NET SDK missing out on including the possible values from enum list. cc: @AlexGhiondea